content/en/tutorials/real-time-multiplayer-colyseus.md

```diff
@@ -0,0 +1,403 @@
+
+```
+> my-app@1.0.0 start
+> ts-node-dev --respawn --transpile-only src/index.ts
+
+✅ development.env loaded.
+✅ Express initialized
+🏟 Your Colyseus App
+⚔️ Listening on ws://localhost:2567
+```
+
+# Including the Colyseus JavaScript SDK
+
+Now we need to add the Colyseus JavaScript SDK on PlayCanvas.
+
+We can do that through a "external script" on PlayCanvas project settings.
+
+Open **"Menu" → "Settings"**:
+
+![settings](/images/tutorials/multiplayer-colyseus/settings.png)
+
+From the Settings panel, expand on **"External Scripts"**, and increase the number of **"URLs"**.
+
+![CDN](/images/tutorials/multiplayer-colyseus/ext_script.png)
+
+In the new **"URL"** field, let's include the Colyseus JavaScript SDK, from a CDN:
+
+```
+https://unpkg.com/colyseus.js@^0.14.0/dist/colyseus.js
+```
+
+This is going to make the `Colyseus` [JavaScript SDK](https://docs.colyseus.io/colyseus/getting-started/javascript-client/) available for our PlayCanvas scripts.
+
+# Establishing a Client-Server Connection
+
+Now, from a new PlayCanvas Script, let's instantiate our `Colyseus.Client` instance. ([see "Creating new scripts"](https://developer.playcanvas.com/en/user-manual/scripting/creating-new/))
+
+You can attach this script to any object in your scene, such as the Camera.
+
+```javascript
+var MainMenu = pc.createScript('mainMenu');
+
+MainMenu.prototype.initialize = function () {
+  //
+  // instantiate the SDK
+  // (no connection is established at this point)
+  //
+  this.app.colyseus = new Colyseus.Client("ws://localhost:2567");
+
+  //
+  // request to join or create a room "my_room"
+  // (establish connection with the server)
+  //
+  this.room = await this.app.colyseus.joinOrCreate("my_room");
+}
+```
+
+> Note that we're using the local `ws://localhost:2567` endpoint here. You need to [deploy your server](https://docs.colyseus.io/arena/getting-started/create-application/) to the public internet in order to play with others online.
+
+When you **"Launch"** your PlayCanvas project now, your client is going to establish a connection with the server, and the server is going to create the room `my_room` on demand for you.
+
+Notice that `my_room` is the default room identifier set by the barebones Colyseus server. You can and should change this identifier in the `arena.config.ts` file.
+
+You must be seeing the following message in your server logs, which means a client successfully joined the room!
+
+```
+19U8WkmoK joined!
+```
+
+# Room State and Schema
+
+In Colyseus, we define shared data through the `Schema` structures.
+
+The state synchronization loop looks like this:
+
+1. State mutations are synchronized automatically from Server → Clients
+2. Clients, by attaching callbacks to their local _read-only_ `Schema` structures, can observe for state mutations and react to it.
+3. Clients can send arbitrary messages to the server - which decides what to do with it - and may mutate the state (Go back to step **1.**)
```

Comment:
There are several technical terms used here that may need explaining:

`Schema` and `mutations`